Review Documentation (#1)
malinibalachandran/automated-self-checkout: docs_src/pipelinebenchmarking.md

> # Benchmark Pipeline

Reviewer: "Computer Vision Pipeline Benchmarking" might be a better title for the page

You can benchmark pipelines with a collection of scripts to get the pipeline performance metrics such as video processing in frame-per-second (FPS), memory usage, power consumption, and so on.

## Prerequisites
Before benchmarking, make sure you [set up the pipeline](./pipelinesetup.md).

> ## Steps to Benchmark Pipeline

Reviewer: If the title changes we can follow this with 
## Steps to benchmark computer vision pipelines

### Step 1: Build the benchmark Docker* images

> Benchmark scripts are containerized inside Docker. The following table lists the commands for various platforms. Choose and run the command corresponding to your platforms or hardware.

Reviewer: ```suggestion
Benchmark scripts are containerized inside Docker. The following table lists the commands for various platforms. Choose and run the command corresponding to your hardware configuration.
```

| Platform                                   | Docker Build Command      | Check Success                                |
| ------------------------------------------ | ------------------------- |----------------------------------------------|

> | Intel platforms with Intel integrated GPUs | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-igt</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:igt`</b> images |

Reviewer: ```suggestion
| Intel integrated GPUs | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-igt</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:igt`</b> images |
```

> | Intel platforms with Intel discrete GPUs   | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-xpu</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:xpu`</b> images |

Reviewer: ```suggestion
| Intel discrete GPUs   | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-xpu</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:xpu`</b> images |
```

---

### Benchmark Specified Number of Pipelines

The primary purpose of benchmarking with a specified number of pipelines is to discover the performance and system requirements for a given use case.

**Example:** 

Here is an example of running benchmarking pipelines with specified number of pipelines:
```bash
sudo ./benchmark.sh --pipelines <number of pipelines> --logdir <output dir>/data --init_duration 30 --duration 120 --platform <core|xeon|dgpu.x> --inputsrc <ex:4k rtsp stream with 10 objects>
```

where, the configurable input parameters include: 
- `--logdir` configures the benchmarking output directory
- `--duration` configures the duration, in number of seconds, the benchmarking will run
- `--init_duration` configures the duration, in number of seconds, to wait for system initialization before the benchmarking metrics or data collection begins

You can run multiple pipeline benchmarking with different configurations before consolidating all pipeline output results.

To get the consolidated pipeline results, run the following `make` command:
```bash
make consolidate ROOT_DIRECTORY=<output dir>
```

> This command will give all the performance metrics among different workload cases given the same root directory specified by `ROOT_DIRECTORY` as shown above.

Author: This sentence is not clear. Does it mean that the command will give the performance metrics of all workload cases that are in the specified root directory?

Reviewer: What it is trying to say is that the consolidate command will consolidate (average) all of the metrics between benchmark runs that exist under the 'ROOT_DIRECTORY'. For example it will consolidate the FPS average between 'benchmark/5_object' and 'benchmark/10_object'.

Author: Rephrased. Please check

Reviewer: That looks good

```

---
### Benchmark Stream Density

> Benchmarking a pipeline can also discover the maximum number of workloads or streams that can be ran in parallel for a given target FPS. This information is useful to determine the hardware required to achieve the desired performance for input sources.

Reviewer: ```suggestion
Benchmarking a pipeline can also discover the maximum number of workloads that can be ran in parallel for a given target FPS. This information is useful to determine the hardware required to achieve the desired performance for the workload.
```

---
## Appendix: Benchmark Helper Scripts

> - `camera-simulator.sh`: This script starts the camera simulator. Create two folders named **camera-simulator** and **sample-media**. Place `camera-simulator.sh` in the **camera-simulator** folder. At the same directory level as the camera-simulator folder, create a folder called sample-media. The `camera-simulator.sh` script will start a simulator for each *.mp4* video that it finds in the **sample-media** folder and will enumerate them as camera_0, camera_1, and so on. Make sure that the path to the `camera-simulator.sh` script is mentioned correctly in the `camera-simulator.sh` script.  

Author: The original text instructs to create a folder called sample-media, and that the script would start a simulator for each mp4 file in that folder. Does the script download the mp4 files automatically or should the user copy or download the files to the sample-media folder?

Reviewer: The camera-simulator will not automatically download media. The user will either need to copy their file into sample-media or run [download_sample_videos.sh](https://github.com/intel-retail/automated-self-checkout/blob/main/benchmark-scripts/download_sample_videos.sh)
